docker-compose/docker-compose.yaml

```diff
@@ -619,31 +621,83 @@ services:
       - 'DATA_SOURCE_NAME=postgres://sg:@codeintel-db:5432/?sslmode=disable'
       - 'PG_EXPORTER_EXTEND_QUERY_PATH=/config/code_intel_queries.yaml'
 
-  # Description: TimescaleDB time-series database for code insights data.
+  # Description: This container will update file permissions on the
+  # `codeinsights` database
+  #
+  # Disk: None
+  # Ports: None
+  #
+  # This container should mount the codeinsights volume, change ownership on the files,
+  # remove any reference to the timescaledb library, and then exit cleanly.
+  # After exiting the `codeinsights-db` service is allowed to start.
+  #
+  codeinsights-chown:
+    container_name: codeinsights-chown
+    image: index.docker.io/sourcegraph/alpine-3.12:insiders@sha256:290c7a045f18b8dab3021e6d5667f751b44c5bd488c408ebd387a802a4bf4876
+    cpus: 0.1
+    mem_limit: '10m'
+    command:
+      - "sh"
+      - "-c"
+      - if [ -d /var/lib/postgresql/data/ ]; then chown -R 999:999 /var/lib/postgresql/data/; sed -r -i "s/[#]*\s*(shared_preload_libraries)\s*=\s*'timescaledb(.*)\'/\1 = '\2'/;s/,'/'/" /var/lib/postgresql/data/pgdata/postgresql.conf; fi
+    restart: "on-failure"
+    networks:
+      - sourcegraph
+    volumes:
+      - 'codeinsights-db:/var/lib/postgresql/data/'
+
+  # Description: PostgreSQL database for code insights data.
   #
   # Disk: 128GB / persistent SSD
   # Network: 1Gbps
   # Ports exposed to other Sourcegraph services: 5432/TCP 9187/TCP
   # Ports exposed to the public internet: none
   #
-  # Note: You should deploy this as a container, do not try to connect it to your external
-  # Postgres deployment (TimescaleDB is a bit special and most hosted Postgres deployments
-  # do not support TimescaleDB, the data here is akin to gitserver's data, where losing it
-  # would be bad but it can be rebuilt given enough time.)
   codeinsights-db:
     container_name: codeinsights-db
-    image: 'index.docker.io/sourcegraph/codeinsights-db:insiders@sha256:82fe7f91dbdca1cd397a80f1023d9da4e9c6af103bc36866d800e3a8279eea00'
+    image: 'index.docker.io/sourcegraph/postgres-12-alpine:135107_2022-03-03_9498a8bd3366@sha256:e26b159dc7c0c47d136886390c899816e669a3c2c1ead689bdad0b610364e45e'
```

Comment:
Thoughts on whether we continue using a custom DB image? We could reuse the standard postgres-12-alpine image (less maintenance for customers running a private registry). Or we can keep building a custom `codeinsights-db` image that wraps our postgres image.

Reply:
I like using the our standard postgres-12-alpine image. there's no good reason unless we are going to add custom config on top of the `soucegraph/postgres-12-alpine` image.